### ISSUE TITLE
Incorrect RF01 failure with _dbt_max_partition

### ISSUE THREAD

**Issue by PaddyAlton** [sqlfluff/sqlfluff#6682]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

RF01 trips on a dbt-templated table that uses the `_dbt_max_partition` templating variable.

Since `_dbt_max_partition` is a templating variable, it cannot be given a reference.

### Expected Behaviour

I would expect `sqlfluff-templater-dbt` to recognise the templating variable and not to trip RF01 in this case.

### Observed Behaviour

RF01 trips on a dbt-templated table that uses the `_dbt_max_partition` templating variable.

### How to reproduce

Simplified example:

```sql
SELECT
    some_table.id,
    arr_element,
FROM
    {{ ref('some_table') }} AS some_table
CROSS JOIN
    UNNEST(some_array) AS arr_element
{% if is_incremental() %}
    WHERE
        -- on incremental runs, overwrite last partition (in case incomplete)
        -- and add new partitions
        some_table.ingestion_date >= DATE(_dbt_max_partition) -- noqa: RF01
{% endif %}
```

Note that `some_table._dbt_max_partition` would fail, because `dbt` replaces the templating variable with a SQL construction when it compiles the final query.

### Dialect

BigQuery

### Version

- sqlfluff: 3.3.1
- sqlfluff-templater-dbt: 3.3.1
- dbt-core: 1.9.2
- dbt-bigquery: 1.9.1

### Configuration

```toml
[tool.sqlfluff.core]
templater = "dbt"
dialect = "bigquery"
exclude_rules = "LT08, L034"

[tool.sqlfluff.rules.aliasing.table]
table = "explicit"
[tool.sqlfluff.rules.aliasing.column]
column = "explicit"

[tool.sqlfluff.rules.capitalisation.keywords]
capitalisation_policy = "upper"
[tool.sqlfluff.rules.capitalisation.identifiers]
capitalisation_policy = "lower"
[tool.sqlfluff.rules.capitalisation.functions]
extended_capitalisation_policy = "upper"
[tool.sqlfluff.rules.capitalisation.literals]
capitalisation_policy = "lower"
[tool.sqlfluff.rules.capitalisation.types]
extended_capitalisation_policy = "upper"

[tool.sqlfluff.rules.convention.select_trailing_comma]
select_clause_trailing_comma = "require"
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
